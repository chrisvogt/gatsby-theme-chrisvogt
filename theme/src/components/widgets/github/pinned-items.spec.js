import React from 'react'
import renderer from 'react-test-renderer'

import PinnedItems from './pinned-items'

const mockPinnedItems = [
  {
    __typename: 'Repository',
    id: 'null-js',
    url: 'https://www.github.com/chrisvogt/null/'
  }
]

describe('Widget/GitHub/PinnedItems', () => {
  describe('snapshots', () => {
    it('matches the placeholder snapshot', () => {
      const tree = renderer.create(<PinnedItems isLoading />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('matches the successful state', () => {
      const tree = renderer
        .create(<PinnedItems pinnedItems={mockPinnedItems} isLoading={false} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
