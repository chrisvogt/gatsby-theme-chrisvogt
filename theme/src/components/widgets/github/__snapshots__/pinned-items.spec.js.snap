// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget/GitHub/PinnedItems snapshots matches the placeholder snapshot 1`] = `
<div
  className="css-g7i32x"
>
  <h3
    className="css-klu22i"
  >
    Pinned Items
  </h3>
  <p
    className="css-1avyp1d"
  >
    Pinned items on my GitHub profile.
  </p>
  <div
    className="css-uz5s83"
  >
    <a
      className="css-1l9an2g"
    >
      <div
        className="css-1yz7e9k"
      >
        <div
          className="show-loading-animation"
        >
          <div
            className="css-z5fbhm"
          >
            <div>
              <div
                className="rect-shape"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": 40,
                    "marginBottom": "2em",
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
            </div>
            <div
              className="css-fjakwv"
            >
              <div
                className="text-block"
                style={
                  {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": 0,
                      "maxHeight": "33.333333333333336%",
                      "width": "97%",
                    }
                  }
                />
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": "0.7em",
                      "maxHeight": "33.333333333333336%",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="text-block"
            style={
              {
                "width": "100%",
              }
            }
          >
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": 0,
                  "maxHeight": "20%",
                  "width": "97%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "100%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "94%",
                }
              }
            />
          </div>
        </div>
      </div>
    </a>
    <a
      className="css-1l9an2g"
    >
      <div
        className="css-1yz7e9k"
      >
        <div
          className="show-loading-animation"
        >
          <div
            className="css-z5fbhm"
          >
            <div>
              <div
                className="rect-shape"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": 40,
                    "marginBottom": "2em",
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
            </div>
            <div
              className="css-fjakwv"
            >
              <div
                className="text-block"
                style={
                  {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": 0,
                      "maxHeight": "33.333333333333336%",
                      "width": "97%",
                    }
                  }
                />
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": "0.7em",
                      "maxHeight": "33.333333333333336%",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="text-block"
            style={
              {
                "width": "100%",
              }
            }
          >
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": 0,
                  "maxHeight": "20%",
                  "width": "97%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "100%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "94%",
                }
              }
            />
          </div>
        </div>
      </div>
    </a>
    <a
      className="css-1l9an2g"
    >
      <div
        className="css-1yz7e9k"
      >
        <div
          className="show-loading-animation"
        >
          <div
            className="css-z5fbhm"
          >
            <div>
              <div
                className="rect-shape"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": 40,
                    "marginBottom": "2em",
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
            </div>
            <div
              className="css-fjakwv"
            >
              <div
                className="text-block"
                style={
                  {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": 0,
                      "maxHeight": "33.333333333333336%",
                      "width": "97%",
                    }
                  }
                />
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": "0.7em",
                      "maxHeight": "33.333333333333336%",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="text-block"
            style={
              {
                "width": "100%",
              }
            }
          >
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": 0,
                  "maxHeight": "20%",
                  "width": "97%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "100%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "94%",
                }
              }
            />
          </div>
        </div>
      </div>
    </a>
    <a
      className="css-1l9an2g"
    >
      <div
        className="css-1yz7e9k"
      >
        <div
          className="show-loading-animation"
        >
          <div
            className="css-z5fbhm"
          >
            <div>
              <div
                className="rect-shape"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": 40,
                    "marginBottom": "2em",
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
            </div>
            <div
              className="css-fjakwv"
            >
              <div
                className="text-block"
                style={
                  {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": 0,
                      "maxHeight": "33.333333333333336%",
                      "width": "97%",
                    }
                  }
                />
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": "0.7em",
                      "maxHeight": "33.333333333333336%",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="text-block"
            style={
              {
                "width": "100%",
              }
            }
          >
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": 0,
                  "maxHeight": "20%",
                  "width": "97%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "100%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "94%",
                }
              }
            />
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`Widget/GitHub/PinnedItems snapshots matches the successful state 1`] = `
<div
  className="css-g7i32x"
>
  <h3
    className="css-klu22i"
  >
    Pinned Items
  </h3>
  <p
    className="css-1avyp1d"
  >
    Pinned items on my GitHub profile.
  </p>
  <div
    className="css-uz5s83"
  >
    <a
      className="css-1l9an2g"
    >
      <div
        className="css-1yz7e9k"
      >
        <div
          className="show-loading-animation"
        >
          <div
            className="css-z5fbhm"
          >
            <div>
              <div
                className="rect-shape"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": 40,
                    "marginBottom": "2em",
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
            </div>
            <div
              className="css-fjakwv"
            >
              <div
                className="text-block"
                style={
                  {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": 0,
                      "maxHeight": "33.333333333333336%",
                      "width": "97%",
                    }
                  }
                />
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": "0.7em",
                      "maxHeight": "33.333333333333336%",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="text-block"
            style={
              {
                "width": "100%",
              }
            }
          >
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": 0,
                  "maxHeight": "20%",
                  "width": "97%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "100%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "94%",
                }
              }
            />
          </div>
        </div>
      </div>
    </a>
    <a
      className="css-1l9an2g"
    >
      <div
        className="css-1yz7e9k"
      >
        <div
          className="show-loading-animation"
        >
          <div
            className="css-z5fbhm"
          >
            <div>
              <div
                className="rect-shape"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": 40,
                    "marginBottom": "2em",
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
            </div>
            <div
              className="css-fjakwv"
            >
              <div
                className="text-block"
                style={
                  {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": 0,
                      "maxHeight": "33.333333333333336%",
                      "width": "97%",
                    }
                  }
                />
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": "0.7em",
                      "maxHeight": "33.333333333333336%",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="text-block"
            style={
              {
                "width": "100%",
              }
            }
          >
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": 0,
                  "maxHeight": "20%",
                  "width": "97%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "100%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "94%",
                }
              }
            />
          </div>
        </div>
      </div>
    </a>
    <a
      className="css-1l9an2g"
    >
      <div
        className="css-1yz7e9k"
      >
        <div
          className="show-loading-animation"
        >
          <div
            className="css-z5fbhm"
          >
            <div>
              <div
                className="rect-shape"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": 40,
                    "marginBottom": "2em",
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
            </div>
            <div
              className="css-fjakwv"
            >
              <div
                className="text-block"
                style={
                  {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": 0,
                      "maxHeight": "33.333333333333336%",
                      "width": "97%",
                    }
                  }
                />
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": "0.7em",
                      "maxHeight": "33.333333333333336%",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="text-block"
            style={
              {
                "width": "100%",
              }
            }
          >
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": 0,
                  "maxHeight": "20%",
                  "width": "97%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "100%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "94%",
                }
              }
            />
          </div>
        </div>
      </div>
    </a>
    <a
      className="css-1l9an2g"
    >
      <div
        className="css-1yz7e9k"
      >
        <div
          className="show-loading-animation"
        >
          <div
            className="css-z5fbhm"
          >
            <div>
              <div
                className="rect-shape"
                style={
                  {
                    "backgroundColor": "#efefef",
                    "height": 40,
                    "marginBottom": "2em",
                    "marginRight": 10,
                    "width": 40,
                  }
                }
              />
            </div>
            <div
              className="css-fjakwv"
            >
              <div
                className="text-block"
                style={
                  {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": 0,
                      "maxHeight": "33.333333333333336%",
                      "width": "97%",
                    }
                  }
                />
                <div
                  className="text-row"
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": "1em",
                      "marginTop": "0.7em",
                      "maxHeight": "33.333333333333336%",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="text-block"
            style={
              {
                "width": "100%",
              }
            }
          >
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": 0,
                  "maxHeight": "20%",
                  "width": "97%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "100%",
                }
              }
            />
            <div
              className="text-row"
              style={
                {
                  "backgroundColor": "#efefef",
                  "height": "1em",
                  "marginTop": "0.7em",
                  "maxHeight": "20%",
                  "width": "94%",
                }
              }
            />
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
